File: client/src/app/entities/insurancecompany.ts
import {User} from './user';

import {DataPage} from '../shared/data-page';
import {Insurancecompanystatus} from './insurancecompanystatus';


export class Insurancecompany {
  id: number;
  code: string;
  name: string;
  contact1: string;
  contact2: string;
  address: string;
  email: string;
  fax: string;
  insurancecompanystatus:Insurancecompanystatus;
  tocreation: string;

  creator: User;

  constructor(id: number = null) {
    this.id = id;
  }
}

export class InsurancecompanyDataPage extends DataPage {
  content: Insurancecompany[];
}

File: server/src/main/java/bit/project/server/entity/Insurancecompany.java
package bit.project.server.entity;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.time.LocalDateTime;


@Data
@Entity
@NoArgsConstructor
@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
public class Insurancecompany {
    @Id
    @GeneratedValue(strategy=GenerationType.IDENTITY)
    private Integer id;

    private String code;

    private String name;

    private String contact1;

    private String contact2;

    @Lob
    private String address;

    private String email;

    private String fax;

    private LocalDateTime tocreation;


    @ManyToOne
    private Insurancecompanystatus insurancecompanystatus;

    @ManyToOne
    @JsonIgnoreProperties({"creator","tocreation","roleList"})
    private User creator;


    public Insurancecompany(Integer id){
        this.id = id;
    }



    public Insurancecompany(Integer id,String code, String name) {
        this.code = code;
        this.name = name;
        this.id = id;

    }


}
